src/OrchardCore/OrchardCore.Localization.Core/LocalizationManager.cs

```diff
@@ -1,66 +1,64 @@
-using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
-using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Extensions.Caching.Memory;
 
-namespace OrchardCore.Localization
+namespace OrchardCore.Localization;
+
+/// <summary>
+/// Represents a manager that manage the localization resources.
+/// </summary>
+public class LocalizationManager : ILocalizationManager
 {
+    private const string CacheKeyPrefix = "CultureDictionary-";
+
+    private static readonly PluralizationRuleDelegate _defaultPluralRule = n => (n != 1 ? 1 : 0);
+
+    private readonly IList<IPluralRuleProvider> _pluralRuleProviders;
+    private readonly IEnumerable<ITranslationProvider> _translationProviders;
+    private readonly IMemoryCache _cache;
+
     /// <summary>
-    /// Represents a manager that manage the localization resources.
+    /// Creates a new instance of <see cref="LocalizationManager"/>.
     /// </summary>
-    public class LocalizationManager : ILocalizationManager
+    /// <param name="pluralRuleProviders">A list of <see cref="IPluralRuleProvider"/>s.</param>
+    /// <param name="translationProviders">The list of available <see cref="ITranslationProvider"/>.</param>
+    /// <param name="cache">The <see cref="IMemoryCache"/>.</param>
+    public LocalizationManager(
+        IEnumerable<IPluralRuleProvider> pluralRuleProviders,
+        IEnumerable<ITranslationProvider> translationProviders,
+        IMemoryCache cache)
     {
-        private const string CacheKeyPrefix = "CultureDictionary-";
-
-        private static readonly PluralizationRuleDelegate _defaultPluralRule = n => (n != 1 ? 1 : 0);
-
-        private readonly IList<IPluralRuleProvider> _pluralRuleProviders;
-        private readonly IEnumerable<ITranslationProvider> _translationProviders;
-        private readonly IMemoryCache _cache;
+        _pluralRuleProviders = pluralRuleProviders.OrderBy(o => o.Order).ToArray();
+        _translationProviders = translationProviders;
+        _cache = cache;
+    }
 
-        /// <summary>
-        /// Creates a new instance of <see cref="LocalizationManager"/>.
-        /// </summary>
-        /// <param name="pluralRuleProviders">A list of <see cref="IPluralRuleProvider"/>s.</param>
-        /// <param name="translationProviders">The list of available <see cref="ITranslationProvider"/>.</param>
-        /// <param name="cache">The <see cref="IMemoryCache"/>.</param>
-        public LocalizationManager(
-            IEnumerable<IPluralRuleProvider> pluralRuleProviders,
-            IEnumerable<ITranslationProvider> translationProviders,
-            IMemoryCache cache)
+    /// <inheritdoc />
+    public async Task<CultureDictionary> GetDictionaryAsync(CultureInfo culture)
+    {
+        var cachedDictionary = await _cache.GetOrCreateAsync(CacheKeyPrefix + culture.Name, async (e) =>
```

Comment:
Sicne we don't want entries to be revoked here (we don't, right? even on memory pressure), we can use a keyed Dictionary and remove the prefix on the key (because there can't be conflicts anymore). I did it recently, I am sure you will recall.

Reply:
Can we use `ConccurentDictionary`?

Reply:
Please also add a comment in the code to remember why we need a ConcurrentDictionary (I will forget myself)